### PR TITLE
MTDSA-15011: Add new "INCORRECT_GOV_TEST_SCENARIO" error

### PR DESCRIPTION
**Pull request by sean-flynn-HMRC**
Please approve but **do NOT merge** yet. 

This PR also includes MTDSA-15012. 

In addition, this PR goes beyond these two tickets: this PR adds "RULE_INCORRECT_GOV_TEST_SCENARIO" to all 24 endpoints in v2.

### CODE REVIEW

File: app/v2/support/DownstreamResponseMappingSupport.scala [hunk 1]
```diff
@@ -27,9 +27,15 @@ trait DownstreamResponseMappingSupport {
...
 
-    lazy val defaultErrorCodeMapping: String => MtdError = { code =>
-      logger.warn(s"[${logContext.controllerName}] [${logContext.endpointName}] - No mapping found for error code $code")
-      InternalError
+    lazy val defaultErrorCodeMapping: String => MtdError = {
+      case "UNMATCHED_STUB_ERROR" =>{
+        logger.warn(s"[${logContext.controllerName}] [${logContext.endpointName}] - No matching stub was found")
+        RuleIncorrectGovTestScenarioError
+      }
+      case code => {
```
Comment (jeremystone) [inline app/v2/support/DownstreamResponseMappingSupport.scala:35]: Could probably do with a test in `DownstreamResponseMappingSupportSpec` to cover this.

**Review by jeremystone — COMMENTED** [review]
Not a full review. Just one comment plus we should put something in the RAML/YAML about this error.

File: app/v2/models/errors/mtdErrors.scala [hunk 2]
```diff
@@ -129,6 +130,13 @@ object UnauthorisedError extends MtdError("CLIENT_OR_AGENT_NOT_AUTHORISED", "The
 
 object InvalidBearerTokenError extends MtdError("UNAUTHORIZED", "Bearer token is missing or not authorized")
 
+//Stub errors
+object RuleIncorrectGovTestScenarioError
+    extends MtdError(
+      code = "RULE_INCORRECT_GOV_TEST_SCENARIO",
+      message = "The Gov-Test-Scenario was not found"
+    )
+
```
Comment (rhysconnor) [inline app/v2/models/errors/mtdErrors.scala:139]: Could we change the format to match the other errors?

File: app/v2/support/DownstreamResponseMappingSupport.scala (hunk 1 above)
Comment (sean-flynn-HMRC) [reply app/v2/support/DownstreamResponseMappingSupport.scala:35]: I believe that's already covered [here](https://github.com/hmrc/property-business-api/pull/361/files#diff-57330e552c06302dea0188f3c119e626d951939bf8201eac35ac94a3afd240a3R62)?

File: app/v2/models/errors/mtdErrors.scala (hunk 2 above)
Comment (sean-flynn-HMRC) [reply app/v2/models/errors/mtdErrors.scala:135]: Yes we can

File: app/v2/support/DownstreamResponseMappingSupport.scala (hunk 1 above)
Comment (jeremystone) [reply app/v2/support/DownstreamResponseMappingSupport.scala:35]: So it is! Cool.

**Review by rhysconnor — APPROVED** [review]
LGTM 🚀  (waiting for confirmation before merging)

**Pull request merged**